users/tests/test_views.py

```diff
@@ -1,13 +1,85 @@
 from django.contrib.auth import get_user_model
 from django.contrib.auth.models import Group, Permission
 from django.test import TestCase
+from django.urls import reverse
+
+from feedbacks.tests.mixins import SetUpUserMixin
+from users.tests_data.test_view_data import REGISTER_DATA, UPDATE_PROFILE_DATA
+
+__all__ = ['LoginViewTest', 'RegisterViewTest', 'UpdateUserViewTest',
+           'PasswordChangeViewTest', 'TestViews']
+
+
+class LoginViewTest(SetUpUserMixin, TestCase):
+
+    def setUp(self):
+        self.client.login(username='test_user', password='12345')
+
+    def test_redirect_when_user_logged_in(self):
+        response = self.client.post(reverse('login'), {'username': 'test_user',
+                                                       'password': '12345'})
+
+        self.assertEqual(response.status_code, 302)
+
+    def test_redirect_to_home_page_if_user_logged_in(self):
+        response = self.client.get(reverse('login'))
+
+        self.assertEqual(response.status_code, 302)
+        self.assertRedirects(response, reverse('home'))
+
+
+class RegisterViewTest(SetUpUserMixin, TestCase):
+
+    def setUp(self):
+        self.client.login(username='test_user', password='12345')
+
+    def test_register_view(self):
+        response = self.client.post(reverse('register'), REGISTER_DATA)
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(get_user_model().objects.get(username='Yurii').email,
+                         'jura@mail.com')
```

Comment:
@MrKreg, it isn't a testing some framework code, it's a testing the end result that's stored in DB after registration. Because 302 code doesn't mean that some model was created(it's a simple redirection), as result this `assert` could be. 
We also can to check redirect page in that test. 
And it would be better to get `username` value from  `REGISTER_DATA` var

Reply:
@rostIvan Django views redirects you only when operation has been finished successful. And saving data into DB it`s framework code that we don`t need to test. There we need only to test response code, redirect to expected route and maybe using of expected template.

Reply:
These line of code really doesn't cost the time for discussion. You doesn't convinced me. But it's ok. I only can to say that full test is better then partially covered.